save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/AvatarStorage.kt

```diff
@@ -30,15 +33,28 @@ class AvatarStorage(configProperties: ConfigProperties) :
         pathToContent.name
     )
 
+    /**
+     * @param key
+     * @param content
+     * @return `Mono` with file size
+     */
+    fun upsert(key: AvatarKey, content: Flux<ByteBuffer>): Mono<Long> {
+        return list()
+            .filter { it.folderUserName == key.folderUserName }
+            .singleOrEmpty()
+            .map { delete(it) }
+            .flatMap { upload(key, content) }
```

Comment:
Here you will execute `upload of key and content` for each and every element from the `list`. If you will have 1000 elements you will do it 1000 times. Are you sure that you really want that? I doubt.

Reply:
Why? I call the _singleOrEmpty()_ method before this

Reply:
Will `flatMap` be called if flux is empty?

Reply:
Yes. Will not. Already checked